### PR TITLE
Refactor population try2

### PR DESCRIPTION
**Pull request by Brett208**
Going to close #1008. Found it safer to manually reapply changes than try to work through the complex merge conflicts. A couple other changes were made in addition.

Related to issue #996

### CODE REVIEW

File: OPHD/Population/PopulationTable.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		Worker,
+		Scientist,
+		Retired
+	};
+
+	int& operator[](std::size_t);
+	int operator[](std::size_t) const;
+
+	int& operator[](Role);
+	int operator[](Role) const;
```
Comment (Brett208) [inline OPHD/Population/PopulationTable.h:21]: @DanRStevens, if you have some time, I would appreciate checking these 4 container access operator overloads to make sure I applied them well. The code is working when testing. Having the extra overloads allows accessing via enum class Role or std::size_t without applying static_casts in code.

**Comment by Brett208** [comment]
Forgot to mention: 

This branch creates a `PopulationTable`, which is very similar to the `ResourcePool` class to separate some of the logic out of population.h. As a side affect, it allowed turning the `enum PopulationRole` into an enum class that I thought worked out really well.

Otherwise, attempted to not change any of the actual game logic within Population.cpp.

**Review by DanRStevens — APPROVED** [review]
Thank you for handling that. That looks much improved.

Wow, there are a lot of uses of the `PersonRole` enum.

I kind of wonder if reducing the use of the `PersonRole` enum could potentially lead to a better design. Such as not needing to explicitly specify a particular role, but rather to have all roles somehow processed together. Though there are significant non-identical behaviours between the population roles, so maybe not.

File: OPHD/Population/Population.h [hunk 2]
```diff
@@ -1,35 +1,24 @@
 #pragma once
 
 #include "Morale.h"
-
-#include <array>
+#include "PopulationTable.h"
 #include <vector>
```
[the remark this answers is not in the document]
Comment (Brett208) [reply OPHD/Population/Population.h:5]: Thanks for the quick review. I will start implementing the vertical space between include blocks.

**Pull request merged**